### DISCUSSION TITLE
Building MINA using NIX for arm64 chip

### DISCUSSION THREAD

**Discussion by ImStaked** [MinaProtocol/mina#13118] [Q&A]
I am attempting to build the mina binary from source using the [instructions found here](https://github.com/MinaProtocol/mina/blob/develop/nix/README.md)
```
git clone mina-repo && cd mina
git checkout 1.4.0beta2
sh <(curl -L https://nixos.org/nix/install) --daemon
mkdir -p "${XDG_CONFIG_HOME-${HOME}/.config}/nix"
echo 'experimental-features = nix-command flakes' > "${XDG_CONFIG_HOME-${HOME}/.config}/nix/nix.conf"
./nix/pin.sh
nix develop mina
nix build mina
```

- Following the instructions above I get a binary that seems to function but only for the dev network.

**Comment by shimkiv** [comment]
Because it defaults to [devtet](https://github.com/MinaProtocol/mina/blob/e39abf79b7fdf96717eb8a8ee88ec42ba1e2663d/nix/ocaml.nix#L150) Dune profile.
While within the develop shell (`nix develop mina`) try build it using [any other profile available](https://github.com/MinaProtocol/mina/tree/develop/src/config) like this: 

```shell
DUNE_PROFILE="mainnet" dune build src/app/cli/src/mina.exe
```

Note thought that since you're going to use artefact built from sources then you will have to also provide Daemon's `--config-file` CLI argument with the path to Genesis which corresponds to [target network](https://github.com/MinaProtocol/mina/tree/develop/genesis_ledgers) (mind the Git branches).
